src/frontend/src/utils/condition.rs

```diff
@@ -382,11 +391,108 @@ impl Condition {
             }
         }
 
+        let (mut row_conjunctions, row_conjunctions_without_struct): (Vec<_>, Vec<_>) =
+            self.conjunctions.clone().into_iter().partition(|expr| {
+                if let Some(f) = expr.as_function_call() {
+                    if let Some(f_input) = f.inputs().get(0)
+                        && let Some(f_input) = f_input.as_function_call()
+                        && matches!(f_input.func_type(), ExprType::Row)
+                    {
+                        true
+                    } else {
+                        false
+                    }
+                } else {
+                    false
+                }
+            });
+
+        // optimize for single row conjunctions. More optimisations may come later
+        // For example, (v1,v2,v3) > (1, 2, 3) means all data from (1, 2, 3).
+        // Suppose v1 v2 v3 are both pk, we can push (v1,v2,v3）> (1,2,3) down to scan
+        // Suppose v1 v2 are both pk, we can push (v1,v2）> (1,2) down to scan and add (v1,v2,v3) > (1,2,3) in filter, it is still possible to reduce the value of scan
+        if row_conjunctions.len() == 1 {
+            let row_conjunction = row_conjunctions.pop().unwrap();
+            let row_left_inputs = row_conjunction
+                .as_function_call()
+                .unwrap()
+                .inputs()
+                .get(0)
+                .unwrap()
+                .as_function_call()
+                .unwrap()
+                .inputs();
+            let row_right_literal = row_conjunction
+                .as_function_call()
+                .unwrap()
+                .inputs()
+                .get(1)
+                .unwrap()
+                .as_literal()
+                .unwrap();
+            let row_right_literal_data = row_right_literal.get_data().clone().unwrap();
+            let row_right_literal_type = row_right_literal.get_data_type().clone().unwrap();
+            let right_iter = row_right_literal_data
+                .as_struct()
```

Comment:
I think we should use `is_struct` to ensure the literal data type is struct as well.